pkg/tsdb/elasticsearch/response_parser.go

```diff
@@ -527,18 +527,19 @@ func (rp *responseParser) trimDatapoints(queryResult backend.DataResponse, targe
 		return
 	}
 
-	trimEdges, err := histogram.Settings.Get("trimEdges").Int()
-	if err != nil {
```

Comment:
`trimEdges` is defined to be a `string` at https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/elasticsearch/components/QueryEditor/BucketAggregationsEditor/aggregations.ts#L20 ,
so i don't think we need to handle any other dataype there.

so i'd recommend to just convert it to string and then do the `atoi` inline, something like 
```go
strconv.Atoi(histogram.Settings.Get("trimEdges").MustString(""))
```

what do you think?

Reply:
Happy to apply the suggestion, I just went the extra round because I noticed that `trimEdges` seems to be used somehow inconsistently in different places. For instance

In this test it is mocked and expected to be a numeric value: https://github.com/grafana/grafana/blob/a58edc9f5e3f8b18772dab48b9396c91109a25c3/pkg/tsdb/elasticsearch/time_series_query_test.go#L1208

Also in this dev dashboard it is exported and stored as an integer:
https://github.com/grafana/grafana/blob/23956557d8c6a119b7de5be5c42024e29634d002/devenv/dev-dashboards/datasource-elasticsearch/elasticsearch_compare.json#L97

I also tested locally by editing the raw JSON of the panel and setting a numeric value directly (instead of using the query editor) an actual number is stored & sent to backend.

Again, happy to simply the code, I was just being extra cautious with the change 😅.

Reply:
@jorgelbg  hmm.. yeah, good point. please don't do it yet, let me dig more into the history of this thing, i'll come back to you 👍

Reply:
@jorgelbg let's go with supporting both modes (number-in-string and really-number) 👍 . i added some more comments below. thanks for finding this complication!